packages/design-picker/src/components/theme-preview/index.tsx

```diff
@@ -58,25 +76,51 @@ const ThemePreview = ( { url, viewportWidth }: Props ) => {
 		};
 	}, [ setIsLoaded, setViewport ] );
 
+	useEffect( () => {
+		iframeRef.current?.contentWindow?.postMessage(
+			{
+				channel: `preview-${ calypso_token }`,
+				type: 'inline-css',
+				inline_css: inlineCss,
+			},
+			'*'
+		);
+	}, [ inlineCss ] );
+
+	const handleDeviceClick = ( device: Device ) => {
+		setDevice( device );
+	};
+
 	return (
 		<div
 			className={ classnames( 'theme-preview__container', {
 				'theme-preview__container--loading': ! isLoaded,
+				'theme-preview__container--frame-bordered': isShowFrameBorder,
+				'theme-preview__container--is-computer': device === 'computer',
+				'theme-preview__container--is-tablet': device === 'tablet',
+				'theme-preview__container--is-phone': device === 'phone',
 			} ) }
 		>
 			{ containerResizeListener }
-			<iframe
-				title={ __( 'Preview', __i18n_text_domain__ ) }
-				className="theme-preview__frame"
-				style={ {
-					width: viewportWidth,
-					height: viewport?.height,
-					transform: `scale(${ scale })`,
-				} }
-				src={ addQueryArgs( url, { calypso_token } ) }
-				scrolling="no"
-				tabIndex={ -1 }
-			/>
+			{ isShowDeviceSwitcher && <Toolbar device={ device } onDeviceClick={ handleDeviceClick } /> }
```

Comment:
[nit] (not sure if you prefer this?)

```suggestion
			{ isShowDeviceSwitcher && <Toolbar device={ device } onDeviceClick={ setDevice } /> }
```

Reply:
Updated! Somehow I missed this 🙈